### ISSUE TITLE
Enable tests on FreeBSD

### ISSUE THREAD

**Issue by casey** [japaric/trust#77]
Why are tests on FreeBSD disabled? I noticed that they fail when enabled, but I figured I would ask here before digging too deep into why.

If they can't be enabled, it might be worth noting why somewhere.

Thanks!

**Comment by japaric** [comment]
It's not possible to do *user* QEMU emulation of *BSD targets which is what we use for the Linux targets.  QEMU *system* emulation would work for this but (a) it's not simple to do "run binary $foo within a QEMU virtual machine and return the exit code of that binary" from the command line, (b) even if that was easy it would be extremely slow to spin one QEMU VM per test binary, (c) there's no out of the box solution for the alternative "spin a single *BSD VM and execute test binaries one by one by sending them via TCP or similar" and (d) even if that was available it's not possible (AFAIK) to automate the creation of the minimal *BSD image that would have to be shipped within the corresponding Docker image that Cross uses to do the technique described in (c), which means that some manual work would be required to create and maintain that *BSD image.

TL;DR It's hard to do and nobody has taken the job to do it. If you are interested there's an implementation of the stuff described in (c) in the rust-lang/rust repo, which is used to run the compiler test suite on an emulated ARM Linux machine.
